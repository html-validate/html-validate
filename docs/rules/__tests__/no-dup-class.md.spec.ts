import HtmlValidate from '../../../build/htmlvalidate';

const markup = {};
markup["incorrect"] = `<div class="foo bar foo"></div>`;
markup["correct"] = `<div class="foo bar"></div>`;

describe('docs/rules/no-dup-class.md', () => {
	it('inline validation: incorrect', () => {
		const htmlvalidate = new HtmlValidate({"rules":{"no-dup-class":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it('inline validation: correct', () => {
		const htmlvalidate = new HtmlValidate({"rules":{"no-dup-class":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
