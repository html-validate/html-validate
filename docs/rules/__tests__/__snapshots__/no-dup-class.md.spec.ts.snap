// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/no-dup-class.md inline validation: correct 1`] = `[]`;

exports[`docs/rules/no-dup-class.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 21,
        "line": 1,
        "message": "Class "foo" duplicated",
        "offset": 20,
        "ruleId": "no-dup-class",
        "ruleUrl": "https://html-validate.org/rules/no-dup-class.html",
        "selector": "div",
        "severity": 2,
        "size": 3,
      },
    ],
    "source": "<div class="foo bar foo"></div>",
    "warningCount": 0,
  },
]
`;
